v1.5 final release: quick hits (#3249)
fishtown-analytics/docs.getdbt.com: website/docs/docs/collaborate/govern/model-contracts.md

## FAQs

> ### Where are contracts supported?

Reviewer: non blocker but this section almost feels like it should be on the top as part of the intro. not really a prerequisites

Author: good call!

At present, model contracts are supported for:
- SQL models (not yet Python)
- Models materialized as `table`, `view`, and `incremental` (with `on_schema_change: append_new_columns`)

> - On the most popular data platforms — but which `constraints` are supported/enforced varies by platform

Reviewer: i'd recommend being more clear on which platforms are supported here. what are 'popular ones'? also should we link to the supported data platform page?

```suggestion
- Most commonly used [data platforms](/docs/supported-data-platforms), such as Snowflake, BigQuery, Databricks, Redshift, and so on, support `constraints`. However, the specific `constraints` that are supported or enforced can vary depending on the platform.
```

### Which models should have contracts?

Any model that meets the criteria above can define a contract.

> Defining contracts for "public" models that are being shared with other groups, teams, and (soon) dbt projects is especially important. For more, read about ["Model access"](model-access).

Reviewer: ```suggestion
It's important to define contracts for "public" models that are shared with other groups, teams, and (soon) dbt projects. Refer to [Model access](/docs/collaborate/govern/model-contracts) for more info.

```

In a parallel for software APIs, the structure of the API response is the contract. Quality and reliability ("uptime") are also very important attributes of an API's quality, but not part of the contract per se, indicating a breaking change or requiring a version bump.

> When you have the option of testing or constraining—the constraint is likely to save on overall cost and compute. The prerequisites for replacing a test with a constraint are:

Reviewer: ```suggestion
When you're deciding between using a test or a constraint, using a constraint can save on overall costs and computing. However, there are some prerequisites you need to meet before replacing a test with a constraint:
```

> - Your data platform supports & enforces the constraint you need. (Most analytical platforms enforce only `not_null`.)

Reviewer: ```suggestion
- Making sure that your data platform can support and enforce the constraint that you need. Most platforms only enforce `not_null`.
```

> - You're materializing as table or incremental (NOT a view or ephemeral)

Reviewer: ```suggestion
- Materializing your data as a table or incremental (not a view or ephemeral)
```

> - You're defining a contract for this model, by declaring every column's `name` + `data_type`.

Reviewer: ```suggestion
- Defining a contract for this model by specifying the `name` and `data_type` of each column.
```
